infra/modules/azure_service_bus_namespace/locals.tf

```diff
@@ -0,0 +1,30 @@
+locals {
+  naming_config = {
+    prefix          = var.environment.prefix,
+    environment     = var.environment.env_short,
+    location        = var.environment.location
+    domain          = var.environment.domain,
+    name            = var.environment.app_name,
+    instance_number = tonumber(var.environment.instance_number),
+  }
+
+  name            = provider::dx::resource_name(merge(local.naming_config, { resource_type = "servicebus_namespace" }))
+  autoscaler_name = replace(local.name, "sbns", "sbns-as")
+
+  sku_name = lookup(
+    {
+      "m" = "Standard",
+      "l" = "Premium"
+    },
+    var.tier,
+    "Premium"
+  )
+
+  capacity       = local.sku_name == "Premium" ? 1 : 0
+  partitions     = local.sku_name == "Premium" ? 1 : 0
+  default_action = local.sku_name == "Premium" ? "Allow" : "Deny" # Using "Deny" for Premium SKU breaks the provider validation
```

Comment:
question: is this a TODO? in case, I'd link the related Jira issue. is there any issue on the provider github repo?

Reply:
No, it is only a provider weird behavior

Reply:
I don't know if it could be filed as bug or not. When you set private endpoints replacing the firewall (available in `Premium` SKU only), the `default_action` variable is useless as the firewall is disabled.
However, provider doesn't allow to set `Deny`, probably because the firewall is actually disabled. More tests would be useful to understand the behavior of the control plane's APIs behind the provider. But honestly, I don't think it does worth doing it though